### PR TITLE
[CT-1103] FNS subaccount WS support

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
websocat "ws://xxx:9091/ws?clobPairIds=1,2&subaccountIds=dydx1nzuttarf5k2j0nug5yzhr6p74t9avehn9hlh8m/0"

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced WebSocket server functionality to dynamically parse and handle `subaccountIds` from incoming requests, improving flexibility and usability.
	- Simplified workflow for building and pushing Docker images, focusing on a single job to enhance maintainability and clarity.
- **Bug Fixes**
	- Implemented robust error handling for missing or incorrectly formatted `subaccountIds`, ensuring smooth operation and user feedback.
	- Added validation to ensure both `clobPairIds` and `subaccountIds` are provided for valid subscription requests.

### CODE REVIEW

File: protocol/streaming/ws/websocket_server.go [hunk 1]
```diff
@@ -91,11 +100,40 @@ func (ws *WebsocketServer) Handler(w http.ResponseWriter, r *http.Request) {
...
+		}
+	}
+
+	return subaccountIds, nil
+}
+
 // parseClobPairIds is a helper function to parse the clobPairIds from the query parameters.
 func parseClobPairIds(r *http.Request) ([]uint32, error) {
 	clobPairIdsParam := r.URL.Query().Get("clobPairIds")
 	if clobPairIdsParam == "" {
```
Comment (jonfung-dydx) [inline protocol/streaming/ws/websocket_server.go:135]: think we can remove this special case now that empty clobpair ids is valid, i think rest of code should result in nil, nil as well

File: protocol/streaming/ws/websocket_server.go [hunk 2]
```diff
@@ -91,11 +100,40 @@ func (ws *WebsocketServer) Handler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// parseSubaccountIds is a helper function to parse the subaccountIds from the query parameters.
+func parseSubaccountIds(r *http.Request) ([]*satypes.SubaccountId, error) {
+	subaccountIdsParam := r.URL.Query().Get("subaccountIds")
+	if subaccountIdsParam == "" {
```
Comment (jonfung-dydx) [inline protocol/streaming/ws/websocket_server.go:106]: ditto here, i think split on empty string probably returns empty array

File: protocol/streaming/ws/websocket_server.go [hunk 3]
```diff
@@ -91,11 +100,40 @@ func (ws *WebsocketServer) Handler(w http.ResponseWriter, r *http.Request) {
...
+		return nil, nil
+	}
+
+	idStrs := strings.Split(subaccountIdsParam, ",")
+	subaccountIds := make([]*satypes.SubaccountId, len(idStrs))
```
Comment (jonfung-dydx) [inline protocol/streaming/ws/websocket_server.go:111]: pref zero-initialize to reduce error situations where an empty array initialized with len X has length X and we might check len(subaccounts) to determine success of parsing subaccount ids

**Review by jonfung-dydx — APPROVED** [review]

File: protocol/streaming/ws/websocket_server.go (hunk 1 above)
Comment (dydxwill) [reply protocol/streaming/ws/websocket_server.go:135]: done

File: protocol/streaming/ws/websocket_server.go (hunk 2 above)
Comment (dydxwill) [reply protocol/streaming/ws/websocket_server.go:106]: done

File: protocol/streaming/ws/websocket_server.go (hunk 3 above)
Comment (dydxwill) [reply protocol/streaming/ws/websocket_server.go:111]: done

**Pull request merged**